### DISCUSSION TITLE
Websocket server hosted from companion?

### DISCUSSION THREAD

**Discussion by gavalierm** [bitfocus/companion#2554] [General]
I have some ESP devices which i want to use as "client" for companion. I like to use websocket.

Nowdays i need to run custom websocket servet (like node.js websocket server) as middleware for ESP device and companion.

This servers do nothing. Only broadcasting the message across devices...

There is a option to use companion as websocket (or other easy and reliable) server to allow thin devices like ESP connect to?

Thank you!

**Comment by Julusian** [comment]
Some modules already do setup their own servers like this.
The challenge is making sure to use a unique port number for each instance of your module.

We don't provide any tooling to help with this, but it shouldn't be any harder than doing it in plain node.js

**Reply by gavalierm** [reply]
Julusian, you know me (a little) i just solving our small needs... ATEM, OBS, Tally, Companion... only thing i need to host extra is plain node.js for websocket/http server hosting (if i want use mobile webapp for tally ..., or for some custom remote webapp for companion..)

So will be great to have websocket server to connect websocets clients.

Do you mean thats will be not implemented ? never? If no i try to find other solution :)
